src/connection_string.ts

```diff
@@ -631,7 +634,28 @@ export const OPTIONS = {
     target: 'credentials',
     transform({ options, values: [value] }): MongoCredentials {
       if (typeof value === 'string') {
-        value = toRecord(value);
+        const validKeys = [
```

Comment:
Are we going to intentionally silently ignore invalid keys here? Does this actually produce the same behavior as before?

Reply:
This array might be unnecessary, the idea was to only translate the type of the keys we know about and leave whatever remains as is (strings). I spoke generally about the options, so that might be where the inspiration came from but I think `CANONICALIZE_HOST_NAME`, and `gssapiCannoncializeHostname` (which we need to add here, whoops!) are the only ones that need translating to boolean. the rest can remain as strings